apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -135,17 +141,17 @@ const WorkshopForm = React.createClass({
     if (this.loadWorkshopRequest) {
       this.loadWorkshopRequest.abort();
     }
-  },
+  }
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.readOnly && !this.props.readOnly) {
-      this.setState(this.getInitialState());
+      this.setState(this.computeInitialState(this.props));
```

Comment:
I'm fairly certain I preserved the original intent and behavior of this code.  Let me know if you disagree.

Reply:
AFAICT this isn't actually needed, because the workshop supplied in `nextProps` here is the same workshop that is already in view. The only change is going from edit mode to view mode, or `readOnly` to not `readOnly`. 

We should probably keep it to be safe, but change it to use `nextProps` instead of the old props. 

In testing this I found an unrelated edit bug where removing multiple sessions on an existing workshop only actually removes the last one. I'll follow up with a fix separately.

Reply:
I verified locally that, aside from the remove sessions bug, this (with `nextProps`) behaves as expected.